packages/ws/package.json

```diff
@@ -18,11 +18,12 @@
 	"typings": "./dist/index.d.ts",
 	"exports": {
 		".": {
+			"types": "./dist/index.d.ts",
 			"import": "./dist/index.mjs",
-			"require": "./dist/index.js",
-			"types": "./dist/index.d.ts"
+			"require": "./dist/index.js"
 		},
 		"./defaultWorker": {
+			"types": "./dist/defaultWorker.d.ts",
```

Comment:
I'm not sure we want this. This file is not meant to be imported - it's just a worker thread script. We have it in our `exports` to ensure users can still manually specify it, like so:
```ts
new WorkerShardingStrategy({ workerPath: '@discordjs/ws/defaultWorker' });
```

Reply:
Isn't this just a different way of importing though? :p

Reply:
No. What I'm saying is, I'd prefer not allowing this:
```ts
import '@discordjs/ws/defaultWorker';
```

though I'm unsure if it currently throws a type error without the `types` field, but I deliberately left it out either way

Reply:
Hm, ye - this case is a little bit convoluted. In theory, `workerPath` wouldn't even have to use `exports` to resolve this thing but I guess that it might be doing that as a convenience. I wonder if it would be capable of loading something outside of `exports`...

> though I'm unsure if it currently throws a type error without the types field, but I deliberately left it out either way

It does throw smth like that today in the [TS playground](https://www.typescriptlang.org/play?moduleResolution=3&target=99&module=1&ts=5.0.4#code/PTAEAEFsHsBMFcA2BTASsgztR8AuBLaAOwC5Qi5kBGANgCgQIYEUyBjaSGIgKwzrr5IAB2gAnXKADeAX1AAzMZ1ABycLHwYOY2H2AB3DMFjJ5AQyS4A6uIDWyMStCNcAT2GYyABgB0ADh8AJgEQoA):
> Cannot find module '@discordjs/ws/defaultWorker' or its corresponding type declarations.(2307)

However, I think that TS playground might be wired differently because the same code **doesn't** error locally and it resolves successfully through the `require` condition and its sibling file (`.d.ts` file next to `.js`, with the same filename)

I feel like you should be able to use `types: null` to "forbid it" at type-level but it currently doesn't work because of the same issue that I'm fixing here ("used fallback condition"). I opened a PR with a failing test case about this in TS: https://github.com/microsoft/TypeScript/pull/54096

To sum it up somehow... this change here doesn't really change how it behaves in TypeScript today. It worked and it still works with it, regardless of the original intention. To fix this today you would have to actively remove `./dist/defaultWorker.d.ts` from disk.

Reply:
a follow up question to this - where do you expect to actually consume this particular entry point? and what's exactly `WorkerShardingStrategy`? what kind of rules govern it?

Reply:
Pretty much, our manager class deals with a bunch of "shards" - websocket connections -  those strategies are used to dictate how to spawn them.

By default, we do in-process, but users can set the strategy to the worker_thread one.

This is where we create the worker threads:
https://github.com/discordjs/discord.js/blob/1b7981e4696b8ae74ceffa876fc0a9be7d0b9f93/packages/ws/src/strategies/sharding/WorkerShardingStrategy.ts#L215

You can see the file path resolution process here:
https://github.com/discordjs/discord.js/blob/1b7981e4696b8ae74ceffa876fc0a9be7d0b9f93/packages/ws/src/strategies/sharding/WorkerShardingStrategy.ts#L236-L256

We already default to that `defaultWorker` file, its in the `exports` field just to ensure the user can also explicitly pass it in.

Reply:
Yeah, that sounds good.